mix.go

```diff
@@ -0,0 +1,50 @@
+package dastard
+
+import (
+	"math"
+)
+
+// Mix performns the mix for lancero data, handles retarding on datastream
+// Retard the raw data stream by 1 sample so it can be mixed with
+//
+// the appropriate error sample. This corrects for a poor choice in the
+// TDM firmware design, but so it goes.
+// fb_physical[n] refers to the feedback signal applied during tick [n]
+// err_physical[n] refers to the error signal measured during tick [n], eg with fb_physical[n] applied
+// fb_data[n]=fb_physical[n+1]
+// err_data[n]=err_physical[n]
+// in words: at frame [n] we get data for the error measured at frame [n]
+// and the feedback that will be applied during frame [n+1]
+// we want
+// mix[n] = fb_physical[n] + mixFraction * err_physical[n]
+// so
+// mix[n] = fb_data[n-1]   + mixFraction * err_data[n]
+// or
+// mix[n+1] = fb_data[n]   + mixFraction * err_data[n+1]
+type Mix struct {
+	mixFraction float64
+	lastFb      RawType
+}
+
+// MixRetardFb mixes err into fbs, alters fbs in place to contain the mixed values
+// consecutive calls must be on consecutive data
+func (m *Mix) MixRetardFb(fbs *[]RawType, errs *[]RawType) {
```

Comment:
I understand that you're not doing this any differently from how I did it, but do we need this copy? I have a sense that we could skip it, that it simplifying the loop but at a cost. Thoughts?

Reply:
I tried to get rid of but failed a few times, I'm not convinced it's impossible though. I stopped working on it because the code is easier to read this way, and it didn't seem like a high priority.